checks_windows.go

```diff
@@ -0,0 +1,410 @@
+	}
+
+	// fmt.Printf("Registry value: %s, keyvalue %s\n", registryValue, keyValue)
+	if registryValue == c.Value {
+		return true, err
+	}
+	return false, err
+}
+
+func (c cond) RegistryKeyExists() (bool, error) {
+	_, err := c.RegistryKey()
+	if err != nil {
+		if err == registry.ErrNotExist {
+			return false, nil
+		} else {
+			return false, err
+		}
+	} else {
+		return true, nil
+	}
+}
+
+func (c cond) ScheduledTaskExists() (bool, error) {
+	return cond{
+		Cmd:   "(Get-ScheduledTask -TaskName '" + c.Name + "').TaskName",
+		Value: c.Name,
+	}.CommandOutput()
+}
+
+func (c cond) SecurityPolicy() (bool, error) {
+	var desiredString string
+
+	// If the passed key is one we know is in the registry, just wrap
+	// RegistryKey.
+	if regKey, ok := secpolToKey[c.Key]; ok {
+		return cond{
+			Key:   regKey,
+			Value: c.Value,
+		}.RegistryKey()
+	}
+
+	// Otherwise, we're going to grab and parse secedit output :/
+	seceditOutput, err := getSecedit()
+	if err != nil {
+		return false, err
+	}
+
+	re := regexp.MustCompile("(?m)[\r\n]+^.*" + c.Key + ".*$")
+	output := strings.TrimSpace(string(re.Find([]byte(seceditOutput))))
+	if output == "" {
+		return false, errors.New("SecurityPolicy item not found")
+	}
+
+	if c.Key == "NewAdministratorName" || c.Key == "NewGuestName" {
+		// These two are strings, not numbers, so they have ""
+		desiredString = c.Key + " = " + c.Value
+	} else if c.Key == "MinimumPasswordAge" ||
+		c.Key == "MinimumPasswordLength" ||
+		c.Key == "LockoutDuration" ||
+		c.Key == "ResetLockoutCount" ||
+		c.Key == "MaximumPasswordAge" ||
+		c.Key == "LockoutBadCount" {
+
+		// These keys are integers, and support ranges.
+		var outputResult, err = strconv.Atoi(strings.Split(output, " = ")[1])
+		if err != nil {
+			return false, err
+		}
+
+		if strings.Contains(c.Value, "-") {
+			splitVal := strings.Split(c.Value, "-")
+			if len(splitVal) != 2 {
+				fail("Malformed range value:", c.Value)
+				return false, errors.New("Invalid c.Value range")
+			}
+			intLow, err := strconv.Atoi(splitVal[0])
+			if err != nil {
+				fail(splitVal[0] + " is not a valid integer for SecurityPolicy check")
+				return false, err
+			}
+			intHigh, err := strconv.Atoi(splitVal[1])
+			if err != nil {
+				fail(splitVal[1] + " is not a valid integer for SecurityPolicy check")
+				return false, err
+			}
+			if intLow <= outputResult && outputResult <= intHigh {
+				return true, nil
+			}
+		} else {
+			desiredValue, err := strconv.Atoi(c.Value)
+			if err != nil {
+				fail(c.Value + " is not a valid integer for SecurityPolicy check")
+				return false, errors.New("Invalid c.Value")
+			}
+			if outputResult == desiredValue {
+				return true, nil
+			}
+		}
+	} else {
+		desiredString = c.Key + " = " + c.Value
+	}
+
+	return output == desiredString, nil
+}
+
+func (c cond) ServiceStartup() (bool, error) {
+	var startupNumber string
+	switch c.Value = strings.ToLower(c.Value); c.Value {
+	case "automatic":
```

Comment:
are 0 and 1 within scope of this project / things likely to be scored with the SE?

Reply:
* for the startupNumbers

Reply:
2/3/4 cover automatic/manual/disabled, so off the top of my head I'm not sure what 0/1 would even mean. Another blessing of the Windows registry.